### PR TITLE
Use comparison P&L csv for download button

### PR DESCRIPTION
**Pull request by AndrewAcomb**
[comparisonProfitAndLoss.csv](https://github.com/user-attachments/files/17141648/comparisonProfitAndLoss.csv)

Example csv above ^

### CODE REVIEW

File: src/components/ProfitAndLossDownloadButton/ProfitAndLossDownloadButton.tsx [hunk 1]
```diff
@@ -15,16 +17,21 @@ export interface PnLDownloadButtonStringOverrides {
 
 export interface ProfitAndLossDownloadButtonProps {
   stringOverrides?: PnLDownloadButtonStringOverrides
+  comparisonConfig?: ProfitAndLossCompareOptionsProps
```
Comment (tomaszantas) [inline src/components/ProfitAndLossDownloadButton/ProfitAndLossDownloadButton.tsx:20]: Perhaps having boolean field instead would be better? Then we don't have to provide whole `ProfitAndLossCompareOptionsProps` to enable this mode

File: src/contexts/ProfitAndLossComparisonContext/ProfitAndLossComparisonContext.tsx [hunk 2]
```diff
@@ -18,4 +19,10 @@ export const PNLComparisonContext = createContext<PNLComparisonContextType>({
   refetch: function (): void {
     throw new Error('Function not implemented.')
   },
+  getProfitAndLossComparisonCsv: function (): Promise<{
+    data?: S3PresignedUrl | undefined
```
Comment (tomaszantas) [inline src/contexts/ProfitAndLossComparisonContext/ProfitAndLossComparisonContext.tsx:23]: Not sure if having both `?` and `undefined` is necessary... however I already experienced some odd behavior and necessity to use both

```suggestion
    data?: S3PresignedUrl
```

File: src/hooks/useProfitAndLossComparison/useProfitAndLossComparison.tsx [hunk 3]
```diff
@@ -42,6 +43,10 @@ type UseProfitAndLossComparison = (props: Props) => {
   compareOptions: TagComparisonOption[]
   setCompareOptions: (options: TagComparisonOption[]) => void
   refetch: (dateRange: DateRange, actAsInitial?: boolean) => void
+  getProfitAndLossComparisonCsv: (
+    dateRange: DateRange,
+    moneyFormat: MoneyFormat | undefined,
```
Comment (tomaszantas) [inline src/hooks/useProfitAndLossComparison/useProfitAndLossComparison.tsx:48]:
```suggestion
    moneyFormat?: MoneyFormat,
```

File: src/hooks/useProfitAndLossComparison/useProfitAndLossComparison.tsx [hunk 4]
```diff
@@ -189,6 +194,25 @@ export const useProfitAndLossComparison: UseProfitAndLossComparison = ({
     ],
   )
 
+  const getProfitAndLossComparisonCsv = (
+    dateRange: DateRange,
+    moneyFormat: MoneyFormat | undefined,
```
Comment (tomaszantas) [inline src/hooks/useProfitAndLossComparison/useProfitAndLossComparison.tsx:199]:
```suggestion
    moneyFormat?: MoneyFormat,
```

**Review by tomaszantas — APPROVED** [review]
Looks good, left few minor suggestions

**Review by ArkuVonSymfon — APPROVED** [review]

**Pull request merged**